hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -282,13 +282,21 @@ case class HoodieResolveReferences(sparkSession: SparkSession) extends Rule[Logi
           // the hoodie's meta field in sql statement, it is a system field, cannot set the value
           // by user.
           if (HoodieSparkUtils.isSpark3) {
-            val assignmentFieldNames = assignments.map(_.key).map {
+            val resolvedAssignments = assignments.map { assign =>
```

Comment:
if spark have not resolve *， assignments will be empty ？ so maybe we have no need to resolve all assignments here。

Reply:
code in line 273 has consider the `empty`. In `HoodieInternalV2Table`,the attribute capabilities contain `ACCEPT_ANY_SCHEMA` which will skip resolve  expression in spark side.
![image](https://user-images.githubusercontent.com/20125927/188581462-c7cff8c0-8872-439d-9990-5045781d2a0f.png)